### PR TITLE
Add new test workflow.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
I decided to extend #118 and add an SQL plugin test workflow. It runs all test jobs regardless of other failures (assuming you don't have compilation errors) and reports all failures after that.
See run example: https://github.com/Bit-Quill/opensearch-project-sql/actions/runs/3077854910 (I intentionally damaged few tests).
Unfortunately, it runs much longer than regular workflow (~22 min), because it repeats some steps, (compilation) multiple times.

_I paused upstream PR while this one is on review_
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/sql-test-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+name: SQL Plugin Tests
+
+on:
+  push:
```
Comment (acarbonetto) [inline .github/workflows/sql-test-workflow.yml:4]: I don't think we need to run this for all branch pushes.  Maybe just for release branches?  This is a big run action with lots of artefacts and could be run manually when needed.

**Review by acarbonetto — CHANGES_REQUESTED** [review]

File: .github/workflows/sql-test-workflow.yml (hunk 1 above)
Comment (Yury-Fridlyand) [reply .github/workflows/sql-test-workflow.yml:4]: I propose to run this workflow only on `dev-*` branches

Comment (acarbonetto) [reply .github/workflows/sql-test-workflow.yml:4]: I was thinking of the opposite.  Do we want to run this on every push?

**Review by MaxKsyunz — APPROVED** [review]

File: .github/workflows/sql-test-workflow.yml (hunk 1 above)
Comment (Yury-Fridlyand) [reply .github/workflows/sql-test-workflow.yml:5]: [Hivemind](https://stackoverflow.com/a/62768214) says
> One thing that's not mentioned in the documentation: the workflow must exist on the default branch for the "Run workflow" button to appear.

I tested - GH prohibits running a workflow with `workflow_dispatch` unless it is on main. I tried GH CLI and web too.
Once it is merged we can update this workflow not to run on `push`. Or not update - depends on demand.

**Review by forestmvey — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**